R/compileModel.R

```diff
@@ -19,23 +19,47 @@ compileModel <- function(model_file, c_file, dll_name, dll_file, hash_file = NUL
     dyn.unload(dll_file)
   }
 
+  # Create a text connection to store output messages generated during the
+  # translation from MCSim model specification text to C.
+  text_conn <- textConnection("mod_output", open = "w")
+
   # Create a C model file (ending with ".c") and an R parameter
   # initialization file (ending with "_inits.R") from the GNU MCSim model
-  # definition file (ending with ".model"). Using the "-R" option generates
-  # code compatible with functions in the R deSolve package.
+  # specification file (ending with ".model"). Using the "-R" option generates
+  # code compatible with functions in the R deSolve package. Write translator
+  # output to the text connection.
+  sink(text_conn)
+  .C("c_mod", model_file, c_file)
+  sink()
+  close(text_conn)
+  mod_output <- paste(mod_output, collapse = "\n")
+
 
-  # if (.C("c_mod", model_file, c_file)[[1]] < 0) {
-  if (is.null(.C("c_mod", model_file, c_file)[[2]])) {
-    stop("c_mod failed")
+  # Check to see if there was an error during translation. If so, print the
+  # translator output and stop execution.
+  if (grepl("Error", mod_output)) {
+    temp_directory <- tempdir()
+    out_file <- file.path(temp_directory, "mod_output.txt")
+    write(mod_output, file = out_file)
+    stop(
```

Comment:
I'm going to write a test for this, will reply back to this message so you can drop it in...

Reply:
tests added in 98475a1